src/DIRAC/DataManagementSystem/Agent/RequestOperations/ReplicateAndRegister.py

```diff
@@ -52,13 +52,20 @@
 from DIRAC.MonitoringSystem.Client.MonitoringReporter import MonitoringReporter
 
 
-def filterReplicas(opFile, logger=None, dataManager=None):
+def filterReplicas(opFile, logger=None, dataManager=None, opSources=None):
     """filter out banned/invalid source SEs"""
 
     if logger is None:
         logger = gLogger
     if dataManager is None:
         dataManager = DataManager()
+    # Create a set of sourceSEs specificed by user
+    userSources = None
+    if opSources:
+        if isinstance(opSources, list):
+            userSources = set(opSources)
+        else:
+            userSources = set([x.strip() for x in opSources.split(",")])
```

Comment:
you could use
 https://github.com/DIRACGrid/DIRAC/blob/3476d79c6c2ec439c90b0da733a80e889042596c/src/DIRAC/RequestManagementSystem/Client/Operation.py#L211
 
 instead of just `sourceSE` when calling `filterReplicas`, so you don't even need to do the strip and split yourself
 
Though I wonder why the hell is the default value `[""]` That clearly seems wrong to me :thinking:

Reply:
I'm happy to use the sourceSEList but that default is a problem... I'd have to do something horrible like `if len(sources) == 1 and not sources[0]: ...` to check if the user has set anything or not!

Would you be happy for the default to be changed to an empty list? At least then it'll evaluate to False if the user hasn't set anything.

Reply:
I believe the current default does not make any sense !